packages/babel-core/test/config-chain.js

```diff
@@ -294,4 +294,196 @@ describe("buildConfigChain", function () {
+    });
+
+    let expected = [
+      {
+        options: {
+          plugins: [
+            "foo",
+            "bar"
+          ]
+        },
+        alias: fixture("js-config", ".babelrc.js"),
+        loc: fixture("js-config", ".babelrc.js"),
+        dirname: fixture("js-config")
+      },
+      {
+        options: {
+          ignore: [
+            "root-ignore"
+          ]
+        },
+        alias: fixture(".babelignore"),
+        loc: fixture(".babelignore"),
+        dirname: fixture()
+      },
+      {
+        options: {
+          filename: fixture("js-config", "src.js"),
+        },
+        alias: "base",
+        loc: "base",
+        dirname: fixture("js-config")
+      }
+    ];
+
+    assert.deepEqual(chain, expected);
+  });
+
+  it("js-config-default", function () {
+    let chain = buildConfigChain({
+      filename: fixture("js-config-default", "src.js")
+    });
+
+    let expected = [
+      {
+        options: {
+          plugins: [
+            "foo",
+            "bar"
+          ]
+        },
+        alias: fixture("js-config-default", ".babelrc.js"),
+        loc: fixture("js-config-default", ".babelrc.js"),
+        dirname: fixture("js-config-default")
+      },
+      {
+        options: {
+          ignore: [
+            "root-ignore"
+          ]
+        },
+        alias: fixture(".babelignore"),
+        loc: fixture(".babelignore"),
+        dirname: fixture()
+      },
+      {
+        options: {
+          filename: fixture("js-config-default", "src.js"),
+        },
+        alias: "base",
+        loc: "base",
+        dirname: fixture("js-config-default")
+      }
+    ];
+
+    assert.deepEqual(chain, expected);
+  });
+
+  it("js-config-extended", function () {
+    let chain = buildConfigChain({
+      filename: fixture("js-config-extended", "src.js")
+    });
+
+    let expected = [
+      {
+        options: {
+          plugins: [
+            "extended"
+          ]
+        },
+        alias: fixture("extended.babelrc.json"),
+        loc: fixture("extended.babelrc.json"),
+        dirname: fixture()
+      },
+      {
+        options: {
+          plugins: [
+            "foo",
+            "bar"
+          ]
+        },
+        alias: fixture("js-config-extended", ".babelrc.js"),
+        loc: fixture("js-config-extended", ".babelrc.js"),
+        dirname: fixture("js-config-extended")
+      },
+      {
+        options: {
+          ignore: [
+            "root-ignore"
+          ]
+        },
+        alias: fixture(".babelignore"),
+        loc: fixture(".babelignore"),
+        dirname: fixture()
+      },
+      {
+        options: {
+          filename: fixture("js-config-extended", "src.js"),
+        },
+        alias: "base",
+        loc: "base",
+        dirname: fixture("js-config-extended")
+      }
+    ];
+
+    assert.deepEqual(chain, expected);
+  });
+
+  it("js-json-config - should use .babelrc if both are present", function () {
```

Comment:
Ok I think we should be erroring if there's both a `.babelrc` and a `.babelrc.js` since there doesn't need to be both at the same level